reasoner/controller/ConjunctionController.java

```diff
@@ -184,80 +194,115 @@ NegatedRequest asNegated() {
             extends AbstractProcessor<ConceptMap, ConceptMap, Request<?>, PROCESSOR> {
 
         final ConceptMap bounds;
-        final List<Resolvable<?>> plan;
+        final ConjunctionStreamPlan plan;
+        final Set<Variable.Retrievable> outputVariables;
+        private final Map<ConjunctionStreamPlan, Map<ConceptMap, PoolingStream.BufferedFanStream<ConceptMap>>> compoundStreamRegistry;
 
         Processor(Driver<PROCESSOR> driver,
                   Driver<? extends ConjunctionController<?, PROCESSOR>> controller,
-                  Context context, ConceptMap bounds, List<Resolvable<?>> plan,
+                  Context context, ConceptMap bounds, ConjunctionStreamPlan conjunctionStreamPlan,
                   Supplier<String> debugName) {
             super(driver, controller, context, debugName);
             this.bounds = bounds;
-            this.plan = plan;
+            this.outputVariables = controller.actor().outputVariables;
+            this.plan = conjunctionStreamPlan;
+            this.compoundStreamRegistry = new HashMap<>();
```

Comment:
Map bounds -> compound stream to reuse sub conjunction results.
CompoundStreams are internal to the processor (and cannot be shared across processors without violating actor isolation).